devito/data/data.py

```diff
@@ -179,9 +179,10 @@ def __str__(self):
         return super(Data, self._local).__str__()
 
     @_check_idx
-    def __getitem__(self, glb_idx, comm_type):
+    def __getitem__(self, glb_idx, comm_type, gather_rank=None):
         loc_idx = self._index_glb_to_loc(glb_idx)
-        if comm_type is index_by_index:
+        gather = True if isinstance(gather_rank, int) else False
```

Comment:
nitpicking for code homogeneity, perhaps for another PR: typically we call these variables "is_..." in this case "is_gather". The "is_" encodes the information that the variable is a boolean